### ISSUE TITLE
final line of multiline talk message from dojo printed out of order

### ISSUE THREAD

**Issue by ault011** [urbit/urbit#601]
This confused the hell out of me. I assume it's an error, but if it's not for some reason it should be...

```
~sivtyv-barnel= is there still tall form dojoo
--------[6.025]
~sivtyv-barnel= quick..
\/~sivtyv-barnel= would look it up in the quickstart if he th\/
  ought it would be
\/ 
```

**Comment by ault011** [comment]
this line  `~sivtyv-barnel= quick..` was obvioulsyt the last line I typed.

**Comment by juped** [comment]
Wow, I can't believe no one had made the "automatically split messages appear out of order" thing an issue yet.

Automatically split messages often appear out of order for some reason, and it looks different for every observer. It's @ohAitch's fault, probably.

**Comment by ohAitch** [comment]
*put them __into__ the arvo event stack in order*, but apparently `:talk` ordering isn't guaranteed...
(@cgyarvin that was the issue right?)

**Comment by ault011** [comment]
ok, if this line splitting thing makes things more complicated then why do
we have it? And if it's not complicated, why haven't we fixed it.

:talk is a key feature to peoples early urbit experience. Misprinting
messages (especially due to a feature that doesn't seem necessary, to me,
at least) is embarrassing.

 Personally, I'm not a fan of the line splitting. That said,  it's more
than possible there's a reason we need it of which I'm not aware.

**Comment by juped** [comment]
One could certainly argue that it is wrong. (I have.) I believe it exists as an attempt at compromise between people who like short messages and people who don't. All compromise, of course, is wrong. :)

**Comment by cgyarvin** [comment]
I'm surprised you're not a fan of the line splitting, Henry!  I normally am
the one who likes coercive restraints on the user.  I don't want people to
have to think about line length if they don't want to.

I don't know what the cause of this bug is, and I think it would warrant
investigation.  All of our messages are ordered, above the packet level.  I
suspect there's an order reversal which is visible only when multiple
messages are bunched in some way.

Basically, Henry, if this annoys you, it should be on Anton's task list...

**Comment by ault011** [comment]
I'll put it there. We can't have new user's first experience with multiline
messages consist of them receiving a messages out of order. It's too sloppy
and easy to fix (hopefully!).

**Comment by jfranklin9000** [comment]
Not sure if this is the same bug but I saw out of order messages between different ships.

`~talkbot` responds to `~nimbus` before being thanked:

````
~piclyn^* but that doesn't really help with what it means to link an app
       ~nimbus/ https://github.com/urbit/docs
~banlyr_fanmud* :: you're welcome!                          <-- swapped
       ~nimbus* (thanks ~talkbot)                           <-- swapped
--------[1.745]
       ~nimbus* very much appreciating pull requests :)
~banlyr_fanmud* :: urbit/docs: urbit documentation          <-- this should appear 3 lines earlier
       ~lasten* if a ticket becomes invalid once a ship is created, how do i go
       ~lasten* about moving a planet/star to a different machine?
~nimbus^* zip up the folder
--------[1.750]
~nimbus^* when the planet/star/moon is stopped
````

Viewed from a different planet the thanks and `~talkbot` response are not out of order:

````
--------[3.760]
~piclyn^+ but that doesn't really help with what it means to link an app
       ~nimbus/ https://github.com/urbit/docs
~banlyr_fanmud+ :: urbit/docs: urbit documentation
       ~nimbus+ very much appreciating pull requests :)
       ~nimbus+ (thanks ~talkbot)
--------[3.765]
~banlyr_fanmud+ :: you're welcome!
       ~lasten+ if a ticket becomes invalid once a ship is created, how do i go
       ~lasten+ about moving a planet/star to a different machine?
~nimbus^* zip up the folder
~nimbus^* when the planet/star/moon is stopped
````

**Comment by Fang-** [comment]
Closing in favor of urbit/arvo#514.
